### PR TITLE
[VITS] Add to TTA pipeline

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
# What does this PR do?

Small changes to the TTA pipeline to allow for the VITS model

### CODE REVIEW

File: src/transformers/pipelines/text_to_audio.py [hunk 1]
```diff
@@ -110,8 +108,10 @@ def _forward(self, model_inputs, **kwargs):
         # we expect some kwargs to be additional tensors which need to be on the right device
         kwargs = self._ensure_tensor_on_device(kwargs, device=self.device)
 
-        # call the generate by defaults or the forward method if the model cannot generate
-        output = self.forward_method(**model_inputs, **kwargs)
+        if self.model.can_generate():
+            output = self.model.generate(**model_inputs, **kwargs)
+        else:
+            output = self.model(**model_inputs, **kwargs)[0]
```
Comment (sanchit-gandhi) [inline src/transformers/pipelines/text_to_audio.py:114]: If we generate, we return a torch tensor as our output.

If we do a forward pass, we return an instance of `ModelOutput`, where the 0-th element is our generated waveform

Hence, we slice here with `[0]` to get our waveform output

File: tests/pipelines/test_pipelines_text_to_audio.py [hunk 2]
```diff
@@ -173,6 +173,32 @@ def test_conversion_additional_tensor(self):
...
         )
 
+    @slow
+    @require_torch
+    def test_vits_model_pt(self):
+        speech_generator = pipeline(task="text-to-audio", model="facebook/mms-tts-eng", framework="pt")
+
+        outputs = speech_generator("This is a test")
+
+        # vits sampling_rate is not straightforward to get
```
Comment (ylacombe) [inline tests/pipelines/test_pipelines_text_to_audio.py:183]: Looks like it is now!
```suggestion
```

**Review by ylacombe — APPROVED** [review]
LGTM ! Thanks for adding this model and correcting the forward case!

File: src/transformers/models/auto/modeling_auto.py [hunk 3]
```diff
@@ -1036,6 +1036,7 @@
         # Model for Text-To-Waveform mapping
         ("bark", "BarkModel"),
         ("musicgen", "MusicgenForConditionalGeneration"),
+        ("vits", "VitsModel"),
```
Comment (ArthurZucker) [inline src/transformers/models/auto/modeling_auto.py:1039]: 💘

File: tests/pipelines/test_pipelines_text_to_audio.py [hunk 4]
```diff
@@ -173,6 +173,31 @@ def test_conversion_additional_tensor(self):
...
+        outputs = speech_generator(["This is a test", "This is a second test"])
+
+        audio = [output["audio"] for output in outputs]
+
+        self.assertEqual([ANY(np.ndarray), ANY(np.ndarray)], audio)
+
+        # test batching
+        outputs = speech_generator(["This is a test", "This is a second test"], batch_size=2)
+
+        self.assertEqual(ANY(np.ndarray), outputs[0]["audio"])
```
Comment (ArthurZucker) [inline tests/pipelines/test_pipelines_text_to_audio.py:199]: Some spaces can be cleaned up 😄

**Review by ArthurZucker — APPROVED** [review]
LGTM thanks !

**Pull request merged**